lib/_stream_readable.js

```diff
@@ -900,20 +899,20 @@ Readable.prototype.wrap = function(stream) {
 
   // proxy certain important events.
   for (var n = 0; n < kProxyEvents.length; n++) {
-    stream.on(kProxyEvents[n], self.emit.bind(self, kProxyEvents[n]));
+    stream.on(kProxyEvents[n], this.emit.bind(this, kProxyEvents[n]));
```

Comment:
On the face of it, looks like `self` was not required here in the first place, the block being directly under the lexical scope of constructor function.